tmt/steps/provision/__init__.py

```diff
@@ -3235,6 +3249,19 @@ def load(self) -> None:
         except tmt.utils.FileError:
             self.debug('Provisioned guests not found.', level=2)
 
+    def _generate_ansible_inventory(self) -> None:
+        """Generate Ansible inventory from provisioned guests."""
+        try:
+            # Get layout from ansible.inventory.layout if it exists
+            layout = self.plan.node.get('ansible', {}).get('inventory', {}).get('layout')
+
+            inventory = self._ansible_inventory.generate(self.guests, layout)
+            inventory_path = Path('inventory.yaml')
```

Comment:
One issue that we were considering at some point is where to put the `inventory.yaml` and other `host_vars`, etc. when those can be exposed. I believe we wanted it under some `TMT_PLAN_DATA` path so that it is accessible for each `prepare` step.

Reply:
This is something I wondered while coding the feature. For now, the path logic and order is 
1. Relative to fmf file
2. Current working dir

Reply:
Let's keep the logic consistent with the [ansible playbook path](https://tmt.readthedocs.io/en/stable/plugins/prepare.html#configuration) handling, that is it should relative to the `fmf` metadata root.

Reply:
@happz @psss thoughts?

Reply:
@LecrisUT This would be also a read-only file, as we are referring to the layout inventory. The output inventory path is fixed and always goes to workdir, in provision folder.

Reply:
I'm using write utils function which writes to the workdir to avoid duplicating code, that's why I don't define the full path at the beginning. I have renamed the variable to make it less confusing